packages/blade/src/components/ListView/_decisions/decisions.md

```diff
@@ -0,0 +1,481 @@
+      {
+        title: 'Pending',
+        onClick: () =>
+          setFilteredTableData(
+            tableDefaultData.filter((tableData) => tableData.status === 'Pending'),
+          ),
+      },
+    ]}
+    filters={[
+      {
+        label: 'Status',
+        options: ['All', 'Pending', 'Completed'],
+        trigger: 'select-input',
+        onChange: ({ name, value }) => {
+          // filter on select
+        },
+      },
+      {
+        label: 'Customer Email',
+        trigger: 'text-input',
+        type: 'email',
+        onBlur: () => {
+          // filter on email change
+        },
+      },
+      {
+        label: 'Date',
+        trigger: 'date-range',
+        onChange: ({ name, value }) => {
+          // filter on date range
+        },
+      },
+    ]}
+    onSearch={(searchValue, searchColumn = 'name') => {
+      setFilteredTableData(
+        tableDefaultData.filter((tableData) => searchValue.includes(tableData[searchColumn])),
+      );
+    }}
+  />
+
+  <Table data={{ nodes: filteredTableData }} />
+</ListView>
+```
+
+**Pros**
+
+- Less verbose compared to using dropdown and menu inside ListViewFilters
+
+**Cons**
+
+- Very strict API which might cause missing edge cases and rework on blade-side for every product change
+- Will have bundle of Dropdown, Menu, DatePicker, BottomSheet, etc even if that particular filter is not used
+- Less intuitive because it'll introduce new config. Compared to previous API where consumers will use filters using the components that they already know how to use like Dropdown, Menu, etc.
+
+#### Extension of Table component
+
+In this case, we add quickfilters, filters, search as an option to our existing Table component
+
+Similar to [react-data-grid](https://mui.com/x/react-data-grid/filtering/quick-filter/) by MUI
+
+```jsx
+<Table
+  data={{ nodes }}
+  hasSearch
+  quickFilters={[
+    { title: 'All', onClick: (nodes) => nodes },
+    {
+      title: 'Pending',
+      onClick: (nodes) => nodes.filter((node) => node.status === 'Pending'),
+    },
+  ]}
+  filters={[
+    {
+      label: 'Status',
+      options: ['All', 'Pending', 'Completed'],
+      trigger: 'select-input',
+      onChange: ({ name, value }) => {
+        // filter on select
+      },
+    },
+    {
+      label: 'Customer Email',
+      trigger: 'text-input',
+      type: 'email',
+      onBlur: () => {
+        // filter on email change
+      },
+    },
+    {
+      label: 'Date',
+      trigger: 'date-range',
+      onChange: ({ name, value }) => {
+        // filter on date range
+      },
+    },
+  ]}
+/>
+```
+
+**Pros**
+
+- Less verbose API
+- Consistent filtering experience on all tables
+
+**Cons**
+
+- Too strict with very little flexibility
+- Adds up more things in tables which can make table difficult to maintain and will contain bundle of filters even when they are not used
+
+</details>
+
+## Enhancements / Components
+
+- Enhancements:
+  - Searchable Dropdown
+  - InputGroup (TBD)
```

Comment:
Where is this needed?

Reply:
It was needed for search but we decided to deprioritise it and do it later
<img width="489" alt="image" src="https://github.com/user-attachments/assets/bfd11776-0a68-4818-906d-6f49246fa1b8" />